ext/calendar/easter.c

```diff
@@ -48,11 +53,16 @@ static void _cal_easter(INTERNAL_FUNCTION_PARAMETERS, bool gm)
 		}
 	}
 
-	if (gm && (year<1970 || year>2037)) {				/* out of range for timestamps */
+	if (gm && (year<1970 || year>2037) && ZEND_LONG_MAX == INT32_MAX) { /* out of range for timestamps on 32bit systems */
 		zend_argument_value_error(1, "must be between 1970 and 2037 (inclusive)");
 		RETURN_THROWS();
 	}
 
+	if (gm && (year<1970 || year>2000000000)) {             /* timestamp range is between 1970 and the year 2.000.000.000 */
+		zend_argument_value_error(1, "must be a year after 1970 (inclusive)");
+		RETURN_THROWS();
+	}
```

Comment:
Split these two checks into individual ``if`` blocks to be able to use a "sensible" wording for years after 1970 without mentioning the upper limit. But if someone does provide a value that hits an upper limit they will get a proper diagnostic.